### ISSUE TITLE
Go to next error/warning/info

### ISSUE THREAD

**Issue by torkelrogstad** [microsoft/vscode#105795]
I'd love a feature where I could go to the next error in the file I'm in, or in the entire project, _but not_ the next warning/info marker. Currently I have these shortcuts available: 

* `editor.action.marker.next`
* `editor.action.marker.nextInFiles`
* `editor.action.marker.prev`
* `editor.action.marker.prevInFiles`

I think it'd make a lot of sense to expose similar shortcuts for going to the next error, or the next warning, or the next info, but as *separate groups*. 

Example use case: I have a spell checker and a linter enabled for my project. The linter produces both warnings and errors, and the spell checker produces info markers. I care mostly about linter errors and compiler errors, and when those are fixed I care about spelling errors.

**Comment by danielo515** [comment]
100% agree
Or if you can focus on the "error provider" of your choice that will be ideal: just next error coming from eslint, next error coming from typescript, etc.

**Comment by mellester** [comment]
It would be nice if we could define something like this
```
    {
        "key": "f4",
        "command": "editor.action.marker.next",
        "args": "Error|Warn"
    }
```
And it would skip the Info markers

**Comment by chrisnevers** [comment]
Are there any plans to implement this feature?

**Comment by rcdailey** [comment]
I don't think this needs extra configuration. Why can't Next/Previous Problem respect the current filter set in the Problems view? Basically if a problem is filtered out, skip it when doing Next Problem keyboard shortcut.

**Comment by danielo515** [comment]
> I don't think this needs extra configuration. Why can't Next/Previous Problem respect the current filter set in the Problems view? Basically if a problem is filtered out, skip it when doing Next Problem keyboard shortcut.

That makes a lot of sense

**Comment by PabloLION** [comment]
IMHO, both adding a new command or passing args to the current one are good.
I searched the market and there's none. Maybe someone can make an extension for this?

**Comment by PabloLION** [comment]
> https://marketplace.visualstudio.com/items?itemName=yy0931.go-to-next-error

I tested this extension and it's quite good! And very thoughtful to write the full guide. 5 stars!
I reckon this issue is ready to be closed for being solved.

**Comment by PabloLION** [comment]
> https://marketplace.visualstudio.com/items?itemName=yy0931.go-to-next-error

I tested this extension by @yy0931 and it's quite good! And very thoughtful to write the full guide. 5 stars!
I reckon this issue is ready to be closed for being solved.

TLDR: please give a heart to this man.

**Comment by Yazir** [comment]
It would be much appreciated if it was already a feature of vscode itself

**Comment by mtone** [comment]
In Jetbrains IDEs "Next Error" loops through all errors across all files, and only when there are no longer any errors, loops through all warnings, etc. I quite like this behaviour and might address or complement the scenario in this issue.

However it doesn't appear possible in VsCode:
- The setting problems.sortOrder = 'Severity' does not do this even when using key "Next problem (all files)". It will prioritize warnings in the current file over errors in others. It was not discussed in https://github.com/microsoft/vscode/issues/135249 where the option was added.
- yy0931's extension requires binding multiple keys to achieve the same.

My typical scenario: adding a required member to the TS interface of a React component props always causes errors in the parent components. No matter how many warnings I have, I'd press F8 to to navigate to those right next -- which might over multiple files -- then keep using F8 to navigate warnings from my current location.

This approach is already suggested in https://github.com/microsoft/vscode/issues/97643.

**Comment by starball5** [comment]
Related Stack Overflow question: [How do I go to the next error (not next problem) in using a keyboard shortcut?](https://stackoverflow.com/q/75451782/11107541)

**Comment by alexpovel** [comment]
As this issue mentions a spell checker as an example, and I found this thread while dealing with that same issue: here's another solution. This pertains to [Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker) aka `streetsidesoftware.code-spell-checker`.  It comes with an option [`diagnosticLevel`](https://github.com/streetsidesoftware/vscode-spell-checker/blob/fafd5425421376b0713119f0bb0890376d594eca/package.json#L2397-L2415), which is `Information` by default, and will subsequently show in the *Problems* pane and be subject to *Go to next error/warning/info* commands. The level can be set to `Hint`, removing the extension's info from the *Problems* pane. *Go to next error/warning/info* then works as expected and doesn't jump to typos anymore.

This is what the *Hint* level then looks like (bottom left):

![image](https://github.com/microsoft/vscode/assets/48824213/84736016-7338-4ba6-be77-685082485732)

So perhaps, as a workaround, see if your extension supports adjusting its reporting level.

**Comment by darianmorat** [comment]
Or also having the option of going to the next error avoiding the message and just jumping directly to the error. I use Error Lens and it shows the msg 2 times.

**Comment by rettoua** [comment]
Any solution for that w/o extension/plugins etc?

**Comment by n-gist** [comment]
try [Go To Problem](https://marketplace.visualstudio.com/items?itemName=Gist.go-to-problem)

It was initially created to hide auto error peek only, but in the process I have met various issues and requests, and checked the work of existing extensions. There are good ones but narrowly focused, or more empowered but implemented in non-standard way of usage. Some are in broken state and have not been updated for a long time. So I tried my best to collect all the highly requested scenarios in one feature-rich extension
https://marketplace.visualstudio.com/items?itemName=Gist.go-to-problem
